### ISSUE TITLE
llab config file needs some changes

### ISSUE THREAD

**Issue by xtitter** [beautyjoy/llab#148]
* it should be json/yaml, rather than .js -- don't want random things executing 
 * it should be called llab-config.yaml or json

 
And, it really should sit  in the root llab directory, sigh. Do we really need to have llab inside the content repo; why can't it be next to it?

**Comment by cycomachead** [comment]
> don't want random things executing

Yes, in theory, but the config file should provide a way to add additional onload functions, that way it's easy to extend llab. For example the Snap! "run" URL's should be in bjc-r's config file not in llab.

> And, it really should sit in the root llab directory, sigh.

But then llab can't be properly extracted as a dependency...or it makes it really hard to work with git and do that. :/

**Comment by cycomachead** [comment]
But I'm also for a JSON or better spec. I don't know what's best. I want to somehow be able to custom define functions. Maybe we allow the config file to specify an external resource to load.

**Comment by xtitter** [comment]
Yeah, pointing to a curriuclum-specific js file would be fine to come up
with a standard way.

I realize no JSON for the config file -- we need comments in there.  YAML.

On Sun, Jul 26, 2015 at 10:47 PM, Michael Ball <notifications@github.com>
wrote:

> But I'm also for a JSON or better spec. I don't know what's best. I want
> to somehow be able to custom define functions. Maybe we allow the config
> file to specify an external resource to load.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beautyjoy/llab/issues/148#issuecomment-125094295>.
>



-- 

  <-><=><-><=><-><=><-><=><-><=><-><=><-><=><-><=><-><=><-><=><->
  Nathaniel Titterton, Ph.D.                    nate@berkeley.edu
  Computer Science Education         U.C. Berkeley, 329 Soda Hall

**Comment by cycomachead** [comment]
> I realize no JSON for the config file -- we need comments in there.  YAML.

Good point. Basically we could support both. YAML gets put into JSON by the browser anyway.

**Comment by xtitter** [comment]
Yeah, except we should hardcode something in loader, or used by the
builder, that finds/loads the config file.  I just made a change to my
local loader.js to find 'llab-config.js', which I know I'm going to screw
up and commit somehow.  hm.

On Sun, Jul 26, 2015 at 11:18 PM, Michael Ball <notifications@github.com>
wrote:

> I realize no JSON for the config file -- we need comments in there. YAML.
>
> Good point. Basically we could support both. YAML gets put into JSON by
> the browser anyway.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beautyjoy/llab/issues/148#issuecomment-125099784>.
>



-- 

  <-><=><-><=><-><=><-><=><-><=><-><=><-><=><-><=><-><=><-><=><->
  Nathaniel Titterton, Ph.D.                    nate@berkeley.edu
  Computer Science Education         U.C. Berkeley, 329 Soda Hall

**Comment by cycomachead** [comment]
> Yeah, except we should hardcode something in loader, or used by the
> builder, that finds/loads the config file.  I just made a change to my
> local loader.js to find 'llab-config.js', which I know I'm going to screw
> up and commit somehow.  hm.

Git branches!

And yeah, well in the browser, we could just try different URLs too...which is sketchy but can work.
